src/main/java/io/supertokens/storageLayer/StorageLayer.java

```diff
@@ -560,4 +565,131 @@ public static List<List<TenantIdentifier>> getTenantsWithUniqueUserPoolId(Main m
+        Storage storage = getStorage(tenantIdentifier, main);
+
+        UserIdMappingStorage uimStorage = (UserIdMappingStorage) storage;
+        AppIdentifier appIdentifier = tenantIdentifier.toAppIdentifier(storage);
+
+        if (userIdType == UserIdType.SUPERTOKENS) {
+            UserIdMapping mapping = uimStorage.getUserIdMapping(
+                    appIdentifier, userId, true);
+            if (mapping != null) {
+                return new TenantIdentifierStorageAndUserIdMapping(
+                        tenantIdentifier.withStorage(storage), storage, mapping);
+            }
+
+        } else if (userIdType == UserIdType.EXTERNAL) {
+            UserIdMapping mapping = uimStorage.getUserIdMapping(
+                    appIdentifier, userId, false);
+            if (mapping != null) {
+                return new TenantIdentifierStorageAndUserIdMapping(
+                        tenantIdentifier.withStorage(storage), storage, mapping);
+            }
+
+        } else {
+            io.supertokens.pluginInterface.useridmapping.UserIdMapping[] userIdMappings = uimStorage.getUserIdMapping(
+                    appIdentifier, userId);
+
+            if (userIdMappings.length == 1) {
+                return new TenantIdentifierStorageAndUserIdMapping(
+                        tenantIdentifier.withStorage(storage), storage, userIdMappings[0]);
+
+            } else if (userIdMappings.length == 2) {
+                for (io.supertokens.pluginInterface.useridmapping.UserIdMapping userIdMapping : userIdMappings) {
+                    if (userIdMapping.superTokensUserId.equals(userId)) {
+                        return new TenantIdentifierStorageAndUserIdMapping(
+                                tenantIdentifier.withStorage(storage), storage, userIdMapping);
+                    }
+                }
+
+            } else if (userIdMappings.length > 2) {
+                throw new IllegalStateException("Retrieved more than 2 UserId Mapping entries for a single userId.");
+            }
+        }
+
+        if (((AuthRecipeStorage) storage).doesUserIdExist(appIdentifier, userId)) {
+            return new TenantIdentifierStorageAndUserIdMapping(
+                    tenantIdentifier.withStorage(storage), storage, null);
+        }
+
+        throw new UnknownUserIdException();
+    }
+
+    public static AppIdentifierStorageAndUserIdMapping getAppIdentifierStorageAndUserIdMappingForUser(
+            Main main, AppIdentifier appIdentifier, String userId,
+            UserIdType userIdType) throws StorageQueryException, TenantOrAppNotFoundException, UnknownUserIdException {
+
+        Storage[] storages = getStoragesForApp(main, appIdentifier);
+
+        if (storages.length == 0) {
+            throw new TenantOrAppNotFoundException(appIdentifier);
+        }
+
+        for (Storage storage : storages) {
+            UserIdMappingStorage uimStorage = (UserIdMappingStorage) storage;
+
```

Comment:
call the user id mapping function instead of replicating the logic

Reply:
Done